tests/baselines/reference/objectLiteralExcessProperties.errors.txt

```diff
@@ -91,4 +101,37 @@ tests/cases/compiler/objectLiteralExcessProperties.ts(33,27): error TS2322: Type
 !!! error TS2322: Type '{ colour: string; }' is not assignable to type 'Cover'.
 !!! error TS2322:   Object literal may only specify known properties, but 'colour' does not exist in type 'Cover'. Did you mean to write 'color'?
 !!! related TS6501 tests/cases/compiler/objectLiteralExcessProperties.ts:28:5: The expected type comes from this index signature.
+    
+    // Repros inspired by #28752
+    
+    function test<T extends IFoo>() {
+                            ~~~~
+!!! error TS2304: Cannot find name 'IFoo'.
+        // No excess property checks on generic types
+        const obj1: T = { name: "test" };
+              ~~~~
+!!! error TS2322: Type '{ name: string; }' is not assignable to type 'T'.

```

Comment:
Side note: I've seen a couple issues opened where people don't really understand constrained generics well and are surprised an assignment like this fails and don't "get" the error. A little more elaboration could help here (ie, when assignment to a bare type parameter fails), like `A is assignable to T's constraint B, but not T itself.`, just to indicate that "we looked at what you're thinking of, and that isn't what's going on here".

Unrelated to this PR, though. Just a remark. @DanielRosenwasser you agree?